extension-manifest-v3/src/background/adblocker.js

```diff
@@ -274,11 +274,17 @@ ${scripts.join('\n\n')}}
   );
 }
 
-async function injectScriptlets(tabId, url) {
+async function injectScriptlets(tabId, url, origin) {
   const { hostname, domain } = parse(url);
   if (!hostname || isPaused(hostname)) {
     return;
   }
+  if (origin) {
+    const originHost = parse(origin).hostname;
```

Comment:
Ah, to use the tabStats. To be honest, I did not consider it, since I think it is less reliable. In case of onCommitted, there is no alternative, because it requires to make a decision based on only the iframe URL. But in case of Safari, this problem does not exist as far as I know, so my feeling is that having a dependency on timing of another browser API make the overall system more brittle.

Reply:
Still, if leave with `details.origin`, this function should take `originHostname` argument, and it should only be parsed for chromium event listener.

Reply:
The previous PR (https://github.com/ghostery/ghostery-extension/pull/1692) did the origin parsing in the onCommitted listener.

But to go one step back. In my option, there are two independent problems that we are trying to address:
1. the concrete problem of navigation events from the onCommitted event
2. potential scenarios on Safari where pausing might fail (but we have no example to reproduce)

I still wonder if scenario 2 can be even reached? As I understand, the injection on Safari is described here:
* MacOS: https://github.com/ghostery/ghostery-extension/blob/b6fb0ed58656340098245fb1a9bab3da2fe80377/extension-manifest-v3/src/manifest.safari-macos.json#L80
* iOS: https://github.com/ghostery/ghostery-extension/blob/b6fb0ed58656340098245fb1a9bab3da2fe80377/extension-manifest-v3/src/manifest.safari-ios.json#L80

If I am not mistaken, it injects only in the main frame. Thus, when sending the message, the sender.url should be the URL of the real page.

In sum, I would prefer https://github.com/ghostery/ghostery-extension/pull/1692 over this PR.

Reply:
Good catch, it should be for `all_frames`, to be the same as it is on Chromium.

Reply:
The original issue shows, that there are some scriptlets injected for subframes.